frontend/demo/component/tabs/tabsheet-lazy-initialization.ts

```diff
@@ -0,0 +1,49 @@
+import 'Frontend/demo/init'; // hidden-source-line
+
+import { html, LitElement } from 'lit';
+import { customElement, state } from 'lit/decorators.js';
+import '@vaadin/tabs';
+import '@vaadin/tabsheet';
+import { applyTheme } from 'Frontend/generated/theme';
+import { TabSheetSelectedChangedEvent } from '@vaadin/tabsheet';
+
+@customElement('tabsheet-lazy-initialization')
+export class Example extends LitElement {
+  protected createRenderRoot() {
+    const root = super.createRenderRoot();
+    // Apply custom theme (only supported if your app uses one)
+    applyTheme(root);
+    return root;
+  }
+
+  // tag::snippet[]
+  @state()
+  private visitedTabs = new Set<number>();
+
+  private selectedTabChanged = (event: TabSheetSelectedChangedEvent) => {
```

Comment:
nit: let's use class methods for event listeners instead of arrow functions.

```suggestion
  private selectedTabChanged(event: TabSheetSelectedChangedEvent) {
```